mlflow/telemetry/parser.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any
+
+from mlflow.telemetry.schemas import (
+    AutologParams,
+    GenaiEvaluateParams,
+    LogModelParams,
+    ModelType,
+)
+
+if TYPE_CHECKING:
+    from mlflow.genai.scorers import Scorer
+
+_logger = logging.getLogger(__name__)
+
+
+class TelemetryParser(ABC):
+    @classmethod
+    @abstractmethod
+    def extract_params(cls, func_name: str, arguments: dict[str, Any]) -> Any:
+        """
+        Extract the parameters from the function call.
+
+        Args:
+            func_name: The full function name.
+            arguments: The arguments passed to the function.
+        """
+
+
+class LogModelParser(TelemetryParser):
+    @classmethod
+    def extract_params(cls, func_name: str, arguments: dict[str, Any]) -> LogModelParams | None:
+        splits = func_name.rsplit(".", 2)
+        if len(splits) != 3:
+            _logger.warning(f"Failed to extract log model params for function {func_name}")
+            return
+        flavor = splits[1]
+
+        # model parameter is the first positional argument
+        model = next(iter(arguments.values()), None)
+        model_type = ModelType.MODEL_PATH if isinstance(model, str) else cls.parse_model_type(model)
+
+        record_params = {"flavor": flavor, "model": model_type.value}
+        for param in [
+            "pip_requirements",
+            "extra_pip_requirements",
+            "code_paths",
+            "params",
+            "metadata",
+        ]:
+            record_params[f"is_{param}_set"] = arguments.get(param) is not None
+
+        return LogModelParams(**record_params)
+
+    @classmethod
+    def parse_model_type(cls, model: Any) -> ModelType:
+        try:
+            from mlflow.pyfunc.model import ChatAgent, ChatModel, PythonModel
+        except ImportError:
+            pass
+        else:
+            if isinstance(model, PythonModel):
+                return ModelType.PYTHON_MODEL
+            elif isinstance(model, ChatModel):
+                return ModelType.CHAT_MODEL
+            elif isinstance(model, ChatAgent):
+                return ModelType.CHAT_AGENT
+
+        try:
+            from mlflow.pyfunc.model import ResponsesAgent
+        except ImportError:
+            pass
+        else:
+            if isinstance(model, ResponsesAgent):
+                return ModelType.RESPONSES_AGENT
+
+        return ModelType.PYTHON_FUNCTION if callable(model) else ModelType.MODEL_OBJECT
+
+
+class AutologParser(TelemetryParser):
+    @classmethod
+    def extract_params(cls, func_name: str, arguments: dict[str, Any]) -> AutologParams | None:
+        splits = func_name.rsplit(".", 2)
+        if len(splits) != 3:
+            _logger.warning(f"Failed to extract autolog params for function {func_name}")
```

Comment:
```suggestion
            _logger.debug(f"Failed to extract autolog params for function {func_name}")
```

We should use `debug` for telemetry-related messages.